### PR TITLE
FIR IDE: fix IgnoreTests.handleTestWithWrongDirective

### PR DESCRIPTION
**Pull request by tgeng**
Whenever this function is called, it means the directive is wrong. The current
check actually prevents the test from failing if it's missing a positive
directive

### CODE REVIEW

**Review by punzki — APPROVED** [review]

**Review by jsjeon — APPROVED** [review]

File: plugins/kotlin/frontend-independent/tests/test/org/jetbrains/kotlin/test/utils/IgnoreTests.kt [hunk 1]
```diff
@@ -126,15 +126,7 @@ object IgnoreTests {
...
-        val directiveIsOutdated = when {
-            !testPasses && directive is EnableOrDisableTestDirective.Enable -> true
-            testPasses && directive is EnableOrDisableTestDirective.Disable -> true
-            else -> false
-        }
-
-        val directiveIsMissing = !directiveIsOutdated
-
-        if (directiveIsMissing && INSERT_DIRECTIVE_AUTOMATICALLY) {
+        if (INSERT_DIRECTIVE_AUTOMATICALLY && directive is EnableOrDisableTestDirective.Enable) {
```
Comment (darthorimar) [inline plugins/kotlin/frontend-independent/tests/test/org/jetbrains/kotlin/test/utils/IgnoreTests.kt:129]: The fix seems to be incorrect as it now handles only the case when the directive is `// FIR_COMPARISON` is used to mark the test as passing, but some tests uses `// FIR_IGNORE` to mark tests as failing, in that case `directive is EnableOrDisableTestDirective.Disable`

**Review by darthorimar — CHANGES_REQUESTED** [review]

File: plugins/kotlin/frontend-independent/tests/test/org/jetbrains/kotlin/test/utils/IgnoreTests.kt (hunk 1 above)
Comment (tgeng) [reply plugins/kotlin/frontend-independent/tests/test/org/jetbrains/kotlin/test/utils/IgnoreTests.kt:129]: This logic is only trying to automatically fix the test if it passes. This aligns with the previous code, which doesn't offer to fix the test by removing `// IGNORE_FIR`.

I can add the ability to remove `// IGNORE_FIR` if that's desirable though.

Comment (tgeng) [reply plugins/kotlin/frontend-independent/tests/test/org/jetbrains/kotlin/test/utils/IgnoreTests.kt:129]: Thanks let me update it.

**Review by darthorimar — APPROVED** [review]

**Comment by darthorimar** [comment]
Merged manually, thanks!

**Pull request closed without merging**